### PR TITLE
Error checking

### LINKED ISSUE #32: Add error checking
Check for a ULI on form submission and `onBlur` of the input and show an error message if necessary.

The form submission should render an `Alert`.

The `onBlur` should render a simple message with the input label (see [USWDS example](https://standards.usa.gov/components/form-controls/#text-input)).

### PR DESCRIPTION
**Pull request by awolfe76**
Depends on #31 

Closes #32 

Because this is a simple, one input, form this renders errors `onBlur` and `onSubmit` the same way:

cd-error

There's definitely a few ways to handle this ... I think it would be worth doing a quick demo tomorrow.

---

Then the success message looks like:

cd-success

### CODE REVIEW

File: check-digit/src/js/Form.jsx [hunk 1]
```diff
@@ -29,13 +36,26 @@ class Form extends Component {
...
+        {this.props.errors.map((error, i) => {
+          return (
+            <span key={i} className="usa-input-error-message" role="alert">
+              {error}
+            </span>
+          )
+        })}
         <input
           id="uli"
+          ref="uli"
```
Comment (wpears) [inline check-digit/src/js/Form.jsx]: Could we use functional `ref` here? The string form is deprecated

File: check-digit/src/js/Form.jsx [hunk 2]
```diff
@@ -29,13 +36,26 @@ class Form extends Component {
       )
 
     return (
-      <form className="usa-grid" id="main-content" onSubmit={this.handleSubmit}>
+      <form
+        className="CheckDigitForm usa-grid"
+        id="main-content"
+        onSubmit={this.handleFormSubmit}
+      >
         <label for="uli">Enter a ULI</label>
```
Comment (wpears) [inline check-digit/src/js/Form.jsx]: Looks like you mean `htmlFor` here

File: check-digit/src/js/App.jsx [hunk 3]
```diff
@@ -1,34 +1,84 @@
...
+      this.setCheckDigit(uli)
+    }
+  }
+
   render() {
-    const uli = this.state.uli
-    const checkDigit = this.state.checkDigit
+    const { checkDigit, errors, isSubmitted } = this.state
 
     return [
```
Comment (wpears) [inline check-digit/src/js/App.jsx]: Missed this before, but these elements all need `key`s

**Comment by awolfe76** [comment]
@wpears - good catch-es! Should be good to go.

**Comment by awolfe76** [comment]
I should have mentioned, some of the later commits change the way the errors are handled. The UI now displays errors `onBlur`:

onblur

And `onSubmit`:

onsubmit

**Pull request merged**